### PR TITLE
Improve address book

### PR DESCRIPTION
**Pull request by marc-aurele-besner**
## Improve address book

- Save address in your address book
- Retrieve these addresses in the “Address book” in Send token action modal

### Screenshots

AccountSummary
AddressBookManagement
SendToken

### CODE REVIEW

File: explorer/src/components/WalletSideKick/AccountPreferencesModal.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      label: '',
+      type: WalletType.subspace,
+    }),
+    [],
+  )
+
+  const addAddressBookFormValidationSchema = useMemo(
+    () =>
+      Yup.object().shape({
+        address: Yup.string()
```
Comment (jfrank-summit) [inline explorer/src/components/WalletSideKick/AccountPreferencesModal.tsx:34]: We should validate it is a valid address.

File: explorer/src/constants/routes.ts [hunk 2]
```diff
@@ -42,14 +41,14 @@ export const EXTERNAL_ROUTES = {
...
-    twitter: 'https://x.com/AutonomysNet',
+    twitter: 'https://twitter.com/NetworkSubspace',
     discord: 'https://discord.gg/subspace-network',
     telegram: 'https://t.me/subspace_network',
     github: 'https://github.com/subspace',
     reddit: 'https://www.reddit.com/r/sub',
     medium: 'https://medium.com/subspace-network',
     youtube: 'https://www.youtube.com/channel/UCojYRCZOtVTJHJXivOYJzeQ',
-    linkedin: 'https://www.linkedin.com/company/autonomys/',
+    linkedin: 'https://www.linkedin.com/company/subspace-blockchain/',
```
Comment (jfrank-summit) [inline explorer/src/constants/routes.ts:51]: Were these changes intended?

File: explorer/src/components/WalletSideKick/AccountPreferencesModal.tsx [hunk 3]
```diff
@@ -36,6 +38,11 @@ export const AccountPreferencesModal: FC<ActionsModalProps> = ({ isOpen, prefere
             addresses.map((a) => a.address),
             'Address already exists in the address book',
           )
+          .test(
+            'is-valid-address',
+            'Invalid address (must be a valid Polkadot or Ethereum address)',
+            (value) => isAddress(value) || isEvmAddress(value),
```
Comment (jfrank-summit) [inline explorer/src/components/WalletSideKick/AccountPreferencesModal.tsx:44]: Are we intending to do EVM transactions in the sidekick?

File: explorer/src/constants/routes.ts (hunk 2 above)
Comment (marc-aurele-besner) [reply explorer/src/constants/routes.ts:51]: Good catch, no, this revert is not intentional 😭

File: explorer/src/components/WalletSideKick/AccountPreferencesModal.tsx (hunk 3 above)
Comment (marc-aurele-besner) [reply explorer/src/components/WalletSideKick/AccountPreferencesModal.tsx]: I was trying to keep the door open but I removed it

**Review by jfrank-summit — APPROVED** [review]
